src/SpecializeBranchedLoops.cpp

```diff
@@ -62,6 +62,16 @@ bool should_extract(Expr e) {
     return true;
 }
 
+bool is_extent_var(const Variable *v) {
+    std::vector<std::string> parts = split_string(v->name, ".");
+    for (size_t i = 0; i < parts.size(); ++i) {
```

Comment:
How about only checking parts 1 through N instead of 0 through N? I think it will still safely work, and it won't falsely consider a user variable named 'extent' to be an extent variable.